docs/source/python/extending2.rst

```diff
@@ -0,0 +1,46 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+.. _extending:
+
+Extending pyarrow
+=================
```

Comment:
The idea is that this file can also contain the documentation about creating your own `ExtensionType` (documentation which is missing at the moment), so therefore I think `extending.rst` would be a good name. 
However, we already have `extending.rst` which is about using the pyarrow C++ / cython APIs. Anybody an idea for another name for this file? ("extending2" is not meant to keep :))

Reply:
Maybe `extending_types.rst`? Or `user_defined_types.rst`, since ultimately `__arrow_array__`, `ExtensionType`, and (eventually) custom Arrow-to-Pandas conversions are all about user-defined types.

Reply:
Thanks, renamed to `extending_types.rst` (the title can still include "user defined types" once we add documentation about that)

Reply:
The original document could also be renamed `extending_cpp.rst` or something.